### PR TITLE
deprecate BitSet and BitVec

### PR DESCRIPTION
**Pull request by Gankra**
I've mirrored them out to crates (bit-vec and bit-set) that build on stable.

(not sure if this actually correctly deprecates them in std)

### CODE REVIEW

**Comment by Gankro** [comment]
cc @rust-lang/libs 

r? @aturon

**Comment by Gankro** [comment]
.. it appears to not have.

*throws up hands in disgust*

**Comment by BurntSushi** [comment]
:+1:

**Comment by Gankro** [comment]
Oh yeah, I meant to write up the motivation for this. I guess I woke up this morning and was just like "well obviously everyone knows why we should do this".

BitVec and BitSet are marked unstable still because they have several outstanding design problems:

* Indexing is a hack based on defining TRUE and FALSE statics
* BitVec and BitSet are highly tangled because BitSet requires more low-level access to do effecient bulk
manipulation of the raw blocks. This suggests we should expose the blocks as an interface. However that would "lock in" the representation of the structure to `Vec<u32>`. Marshalling through bits and bytes is a lot of work.
* Misc uncertainties over panic semantics
* Misc uncertainties over naming

None of these are a *real* problem in a crate where we can expose janky interfaces and rename whimsically. 

While I agree that a collection of bits is a very useful datastructure, I'd argue the standard collections library should be small, focusing on The Big Dogs of datastructuring. Vec, HashMap, SortedMap, RingBuf, LinkedList.

**Comment by alexcrichton** [comment]
I'm all for this!

> .. it appears to not have.

You'll need to put the `#[deprecated]` annotations on the source instead of the reexport (e.g. in libcollections)

**Comment by aturon** [comment]
(Note for those following at home: this is blocked on https://github.com/rust-lang/rust/issues/26040, which @Gankro has a PR for that will be reviewed shortly.)

File: src/librustdoc/html/static/main.js [hunk 1]
```diff
@@ -81,12 +81,11 @@
             return;
         }
 
-        if (e.which === 191) { // question mark
```
Comment (tbu-) [inline src/librustdoc/html/static/main.js]: Is this a rebase gone wrong?

Comment (Gankro) [inline src/librustdoc/html/static/main.js]: haha, whoops!

**Comment by aturon** [comment]
Is this ready to go now that the stability lint has been fixed? Probably worth testing a fresh/rebased build to make sure it's working properly.

**Comment by Gankro** [comment]
updated

**Comment by Gankro** [comment]
cc @jroesch for where we still use these

**Comment by jroesch** [comment]
@Gankro the only remaining place is the one we talked about on IRC earlier for attributes. I think if we just move a copy of the implementation into librustc_datastructures we can allow the deprecation then removal of them.

**Comment by Gankro** [comment]
@jroesch I also found it in librustc_lint. I missed it before as the compile error'd out on the first one and stopped.

**Pull request merged**